test/uiSchema_test.js

```diff
@@ -247,6 +247,112 @@ describe("uiSchema", () => {
+      });
+
+      it("should focus on integer input, updown widget", () => {
+        shouldFocus({type: "integer"}, {"ui:widget": "updown", "ui:autoFocus": true});
+      });
+
+      it("should focus on integer input, range widget", () => {
+        shouldFocus({type: "integer"}, {"ui:widget": "range", "ui:autoFocus": true});
+      });
+
+      it("should focus on integer enum input", () => {
+        shouldFocus({type: "integer", enum: [1, 2, 3]}, {"ui:autoFocus": true}, "select");
+      });
+    });
+
+    describe("string", () => {
+      it("should focus on text input", () => {
+        shouldFocus({type: "string"}, {"ui:autoFocus": true});
+      });
+
+      it("should focus on textarea", () => {
+        shouldFocus({type: "string"}, {"ui:widget": "textarea", "ui:autoFocus": true}, "textarea");
+      });
+
+      it("should focus on password input", () => {
+        shouldFocus({type: "string"}, {"ui:widget": "password", "ui:autoFocus": true});
+      });
+
+      it("should focus on color input", () => {
+        shouldFocus({type: "string"}, {"ui:widget": "color", "ui:autoFocus": true});
+      });
+
+      it("should focus on email input", () => {
+        shouldFocus({type: "string", format: "email"}, {"ui:autoFocus": true});
+      });
+
+      it("should focus on uri input", () => {
+        shouldFocus({type: "string", format: "uri"}, {"ui:autoFocus": true});
+      });
+
+      it("should focus on data-url input", () => {
+        shouldFocus({type: "string", format: "data-url"}, {"ui:autoFocus": true});
+      });
+    });
+
+    describe("object", () => {
+      it("should focus on date input", () => {
+        shouldFocus({type: "string", format: "date"}, {"ui:autoFocus": true});
+      });
+
+      it("should focus on date-time input", () => {
+        shouldFocus({type: "string", format: "date-time"}, {"ui:autoFocus": true});
+      });
+
+      it("should focus on alt-date input", () => {
+        shouldFocus({type: "string", format: "date"}, {"ui:widget": "alt-date",  "ui:autoFocus": true}, "select");
+      });
+
+      it("should focus on alt-date-time input", () => {
+        shouldFocus({type: "string", format: "date-time"}, {"ui:widget": "alt-datetime",  "ui:autoFocus": true}, "select");
+      });
+    });
+
+    describe("array", () => {
+      it("should focus on multiple files input", () => {
+        shouldFocus({type: "array", items: {type: "string", format: "data-url"}}, {"ui:autoFocus": true});
+      });
+
+      it("should focus on first item of a list of strings", () => {
+        shouldFocus({type: "array", items: {type: "string", default: "foo"}}, {"ui:autoFocus": true}, "input", ["foo", "bar"]);
+      });
+
+      it("should focus on first item of a multiple choices list", () => {
+        shouldFocus({type: "array", items: {type: "string", enum: ["foo", "bar"]}, uniqueItems: true}, {"ui:widget": "checkboxes", "ui:autoFocus": true}, "input", ["bar"]);
```

Comment:
We usually try to not exceed 120c. line length in this project (maybe we should enforce this rule in eslint config)